perceval/algorithm/tomography/tomography_mle.py

```diff
@@ -0,0 +1,400 @@
+                        eigenvector_list.append(k1 // (2 ** j1))
+                        k1 = k1 % (2 ** j1)
+                    X[eigenvector_index] = np.kron(X[eigenvector_index],
+                                                   B[2 * pauli_list[qubit_index] + eigenvector_list[qubit_index]])
+            P += X
+        return P
+
+    def _povm_operator(self) -> list:
+        # Gives a POVM (positive operator value measure) set suited for tomography
+        B = self._povm_state()
+        L = []
+        for state in B:
+            L.append(_state_to_dens_matrix(state) / (3 ** self._nqubit))
+        return L
+
+    def _compute_matrix(self, j: int) -> np.ndarray:
+        if j == 0:
+            return np.eye((2), dtype='complex_')
+        if j == 1:
+            return (1 / np.sqrt(2)) * np.array([[1, 1], [1, -1]], dtype='complex_')
+        if j == 2:
+            return (1 / np.sqrt(2)) * np.array([[1, 1], [1j, -1j]], dtype='complex_')
+
+    def _input_basis(self) -> list:
+        # Computes input density matrix basis (similar to POVM but not same order)
+
+        B = []
+        for j in range(6 ** self._nqubit):
+            k = []
+            for m in range(self._nqubit - 1, -1, -1):
+                k.append(j // (6 ** m))
+                j = j % (6 ** m)
+            k.reverse()
+            M = 1
+            v = 1
+            for i in k:
+                M = np.kron(self._compute_matrix(i // 2), M)
+
+                if i % 2 == 0:
+                    v = np.kron(np.array([[1], [0]], dtype='complex_'), v)
+                else:
+                    v = np.kron(np.array([[0], [1]], dtype='complex_'), v)
+            B.append(_state_to_dens_matrix(np.dot(M, v)))
+        return B
+
+    @staticmethod
+    def _frobenius_inner_product(A: np.ndarray, B: np.ndarray) -> float:
+        # calculates the inner product associated to Frobenius norm
+        return np.trace(np.dot(np.transpose(np.conjugate(A)), B))
+
+    @staticmethod
+    def _proj_simplex(eigenvalues: list) -> list:
+        # Projects a given real eigen-spectra (sorted in descending order) on positive elements with their sum equal to 1
+        u = 0
+        for j in range(1, len(eigenvalues) + 1):
+            x = eigenvalues[j - 1] - (1 / j) * (sum(eigenvalues[:j]) - 1)
+            if x > 0:
+                u = j
+        if u == 0:
+            w = 0
+        else:
+            w = (1 / u) * (sum(eigenvalues[:u]) - 1)
+        L = []
+        for lambda_i in eigenvalues:
+            L.append(max(lambda_i - w, 0))
+        return L
+
+    @staticmethod
+    def _proj(h_matrix: np.ndarray) -> np.ndarray:
+        # Projects a given hermitian matrix on the cone of positive semi-definite trace=1 matrices
+        eigenvalues, eigenvectors = np.linalg.eigh(h_matrix)
+        eigenvalues2 = list(eigenvalues)
+        eigenvalues2.reverse()
+        L = TomographyMLE._proj_simplex(eigenvalues2)
```

Comment:
You should use the numpy method to reverse ndarray elements
`eigenvalues2 = np.flip(eigenvalues)`
but instead of instantiate `eigenvalues2`
you can call `L = TomographyMLE._proj_simplex(np.flip(eigenvalues))`
and below use `len(eigenvalues)`